hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsNode.java

```diff
@@ -93,20 +88,20 @@ private ExecutorInfo createExecutor(String taskIdName, String nodeName, String e
       .setName(nodeName + " executor")
       .setExecutorId(ExecutorInfoBuilder.createExecutorId("executor." + taskIdName))
       .addAllResources(getExecutorResources())
-      .setCommand(CommandInfoBuilder.createCmdInfo(cmd, getCmdUriList(), getExecutorEnvironment()))
+      .setCommand(CommandInfoBuilder.createCmdInfo(cmd, getCmdUriList(nodeId), getExecutorEnvironment()))
       .build();
   }
 
-  private List<CommandInfo.URI> getCmdUriList() {
+  private List<CommandInfo.URI> getCmdUriList(String nodeId) {
```

Comment:
NodeId is based on the task type in your code, see: nodeId = taskName; So instead of using this complicated logic instead, you should just get task name from the Persistent or LiveState.